cdap-api/src/main/java/io/cdap/cdap/api/app/ErrorCodeProvider.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.api.app;
+
+public interface ErrorCodeProvider {
+
+  // This can be any grouping of errors, not necessarily the phase of execution of a pipeline
+  enum ErrorGroup {
+    UNKNOWN,
+    CONFIGURATION,
+    PROVISIONING,
+    LAUNCH,
+    GCS,
+    DATAPROC,
+    DEPROVISIONING
+  }
+
+  ErrorGroup getErrorGroup();
+
+  String getErrorCode();
+
+  // Defining constants that can be re-used for error codes. Defining an enum does not allow other strings to be sent
+  // using the same errocode field such as API response codes.
+  String NO_NETWORK_CONNECTIVITY = "error-no-network-connectivity";
```

Comment:
'error-' as a prefix seems redundant

Reply:
I don't think it makes sense to have any of these constants in this API class as they are not generic errors, but specific to whatever is throwing the error

Reply:
I was worried about having a long list of errors from everywhere here. But we will probably document the errors, with possible actions user may take. It may be useful to have all errors in one place.

Agree 'error-' prefix is not adding additional info. I remember seeing err or similar prefixes in error code documentation elsewhere. But I think it is good to remove here.